engine/src/main/java/org/terasology/world/chunks/ChunkLightMergingSystem.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.chunks;
+
+import com.google.common.collect.Sets;
+import org.joml.Vector3ic;
+import org.terasology.engine.subsystem.common.ThreadManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.In;
+import org.terasology.world.block.BlockRegion;
+import org.terasology.world.chunks.event.BeforeChunkUnload;
+import org.terasology.world.chunks.event.OnChunkLoaded;
+import org.terasology.world.propagation.light.LightMerger;
+
+import java.util.Set;
+
+/**
+ * System which will process light merging after chunk is ready.
+ */
+@RegisterSystem
+public class ChunkLightMergingSystem extends BaseComponentSystem {
```

Comment:
Oh, we don't even hook that up to the magic chunk generation off-thread process, but it is just a regular system :nerd_face: neat!